src/inmanta/server/services/userservice.py

```diff
@@ -0,0 +1,125 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import logging
+
+import asyncpg
+
+import nacl.exceptions
+import nacl.pwhash
+from inmanta import const, data, protocol
+from inmanta.data import AuthMethod, model
+from inmanta.protocol import common, exceptions
+from inmanta.server import SLICE_DATABASE, SLICE_TRANSPORT, SLICE_USER
+from inmanta.server import config as server_config
+from inmanta.server import protocol as server_protocol
+
+LOGGER = logging.getLogger(__name__)
+
+
+def verify_authentication_enabled() -> None:
+    """raises an UnauthorizedException exception if server authentication is not enabled"""
+    if not server_config.server_enable_auth.get():
+        raise exceptions.UnauthorizedException(
```

Comment:
Shouldn't this be a `BadRequest`? An UnauthorizedException means that the user didn't provide credentials, which is not really the problem here. BadRequest indicates that the client made an error. That is applicable here, because the client shouldn't make these API calls the authentication is disabled on the server.